src/mca/pcompress/zlib/compress_zlib.c

```diff
@@ -99,7 +101,7 @@ static bool zlib_compress(const uint8_t *inbytes, size_t inlen, uint8_t **outbyt
 
     rc = deflate(&strm, Z_FINISH);
     (void) deflateEnd(&strm);
-    if (Z_OK != rc && Z_STREAM_END != rc) {
+    if (Z_STREAM_END != rc) {
```

Comment:
I guess you are saying that `deflate` must either return `Z_STREAM_END` or else some error - and thus, cannot return `Z_OK`? I'm not seeing that in the documentation - can you point to where this is explained?

Reply:
You can see the documentation here: https://github.com/madler/zlib/blob/master/zlib.h#L326-L332 (with relevant lines highlighted)